utils/recursivelyCreateRecommendedMaterial.ts

```diff
@@ -0,0 +1,42 @@
+import { materialDefSchema } from "../validation";
+import {
+  db,
+  materialTable,
+  materialCategoriesTable,
+  materialRecommendationsTable,
+} from "../drizzle";
+
+export async function recursivelyCreateRecommendedMaterial(
+  materialId: string,
+  recommendedMaterials: unknown[]
+): Promise<void> {
+  for (const recommendedMaterial of recommendedMaterials) {
+    const parsedRecommendedMaterial =
+      materialDefSchema.parse(recommendedMaterial);
```

Comment:
Zod schemas generally should not be used in utils. The argument should be typed on line 11, and the parsing should be done when the data first arrives in the API. Parsing is not particularly efficient, so it should be minimized.